content/blog/nats-server-29-release.md

```diff
@@ -0,0 +1,502 @@
+
+In 2.9, the distribution algorithm was tweaked to fairly distribute messages across pending fetch requests.
+
+<img src="/img/nats-server-29-release/pull-fetch-29.jpg" style="display: block; margin: 0 auto 20px auto" width="500px" />
+
+For a set of fetch requests queued on the server, messages will be round-robin distributed until each fetch buffer is full or there are no more pending messages.
+
+#### Inactive threshold for durable consumers
+
+Consumers can be either ephemeral or durable. By design, an ephemeral consumer is intended to be used for one-off reads/replay of a stream and has a limited lifetime. In the client libraries, ephemerals are often created using one of the *convenience* methods. For example, in Go, using the [`JetStream`](https://pkg.go.dev/github.com/nats-io/nats.go#JetStream) interface.
+
+```go
+sub, _ := js.SubscribeSync("events.*")
+// Do some work...
+sub.Unsubscribe()
+```
+
+The client library handles:
+
+- Looking up the stream by pattern matching the subject being subscribed to
+- Auto-creating an *unnamed* consumer (which makes an ephemeral)
+- Initializing a subscription and binding it to the consumer
+
+From the server's perspective, it keeps track of whether at least one subscription is bound to this ephemeral consumer _and_ if the subscription is still _active_. If/when it detects the subscription is no longer active, it will automatically delete the ephemeral consumer after the [`InactiveThreshold`](https://docs.nats.io/nats-concepts/jetstream/consumers#inactivethreshold) period (which defaults to five seconds).
+
+In contrast, durables are intended to survive process crashes, restarts (client or server), and regardless if there are any active subscriptions. The lifecycle of a durable is expected to be managed by the application/operator.
+
+However, what happens if consumers are being created and not properly cleaned up? Each consumer has a small amount of state associated with it that the server needs to keep track of, replicate, restore, etc. To prevent an evergrowing number of consumers, this consumer configuration can now be applied to durables.
+
+```go
+js.AddConsumer("EVENTS", &nats.ConsumerConfig{
+  Durable: "event-processor",
+  AckPolicy: nats.AckExplicitPolicy,
+  InactiveThreshold: time.Hour,
+})
+```
+
+In practice, an inactive threshold for a durable should be set to a value that would be highly unlikely the consumer would *not* have been active in the meantime. In other words, if the consumer is expected to be active regularly every couple minutes, if there is no activity for an hour, that may be a good indicator that the consumer can be auto-deleted.
+
+#### Memory usage for containered deployments
+
+As noted above, the NATS server can be deployed in many different environments. An increasingly common one is in containerized environments where, historically, reliable resource management from within a container has been difficult.
```

Comment:
appended some more details to this paragraph:

```suggestion
As noted above, the NATS server can be deployed in many different environments. An increasingly common one is in containerized environments where, historically, reliable resource management from within a container has been difficult since the Go runtime is unaware of the memory limits set by cgroups in a container and instead base the behavior of the Go GC on the available memory from the host.
```